pagopa/io-app
# fix(IT Wallet): [SIW-1795] Problem with SPID Login Using Custom Tab

> [!WARNING]
> Depends on #6455

## Short description
This PR addresses the issue encountered by some users when logging in via SPID for PID issuance on ITWallet, specifically related to the opening of the `CustomTab`.
Unified the authorization flows for SPID, Cie + PIN, and CieID. Previously, SPID and CieID had a unique flow, while Cie + PIN followed two separate flows (`startAuthFlow` and `completeAuthFlow`). Replaced the opening of a `CustomTab` with a `WebView` for SPID authentication.

## List of changes proposed in this pull request
- Updated `io-react-native-wallet` to 0.26.0 ( authentication outside from the package)
- Created a new screen `ItwSpidIdpLoginScreen` to handle the SPID authentication process within a `WebView`. This screen loads the `authUrl` generated by `startAuthFlow` and sends the `authRedirectURL` back to the state machine when it contains `itWalletIssuanceRedirectUri`
- Refactored the XState machine states related to eID issuance for CieID and SPID, aligning them with the existing flow for CIE + PIN
- Added `openUrlAndListenForAuthRedirect`, which opens the `authUrl` in the browser for CieID and provides the `authRedirectURL`
- Created a new actor `getAuthRedirectUrl` used in the `CieIDBuildAuthRedirectUrl` state of CieID, which utilizes `openUrlAndListenForAuthRedirect` to complete the authentication

## How to test
To test the changes, try obtaining the eID using all three authentication methods (SPID, Cie + PIN, and CieID) and ensure that the eID is successfully obtained in each case. After successfully obtaining the eID, proceed to obtain a credential and ensure that the process completes without issues.


https://github.com/user-attachments/assets/2e9d6a20-bee1-470f-860e-3e6b96c9ca1f
